ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -97,44 +98,47 @@ internal sealed class Startup {
 		// Add support for default root path redirection (GET / -> GET /index.html), must come before static files
 		app.UseDefaultFiles();
 
-		// Add support for static files (e.g. HTML, CSS and JS from IPC GUI)
-		app.UseStaticFiles(
-			new StaticFileOptions {
-				OnPrepareResponse = static context => {
-					if (context.File is { Exists: true, IsDirectory: false } && !string.IsNullOrEmpty(context.File.Name)) {
-						string extension = Path.GetExtension(context.File.Name);
-
-						CacheControlHeaderValue cacheControl = new();
-
-						switch (extension.ToUpperInvariant()) {
-							case ".CSS" or ".JS":
-								// Add support for SRI-protected static files
-								// SRI requires from us to notify the caller (especially proxy) to avoid modifying the data
-								cacheControl.NoTransform = true;
-
-								goto default;
-							default:
-								// Instruct the caller to always ask us first about every file it requests
-								// Contrary to the name, this doesn't prevent client from caching, but rather informs it that it must verify with us first that his cache is still up-to-date
-								// This is used to handle ASF and user updates to WWW root, we don't want from the client to ever use outdated scripts
-								cacheControl.NoCache = true;
-
-								// All static files are public by definition, we don't have any authorization here
-								cacheControl.Public = true;
-
-								break;
-						}
+#if !NETFRAMEWORK && !NETSTANDARD
+		List<string> staticFilesDirectorys = new();
 
-						ResponseHeaders headers = context.Context.Response.GetTypedHeaders();
+		string pluginsPath = Path.Combine(SharedInfo.HomeDirectory, SharedInfo.PluginsDirectory);
```

Comment:
You're duplicating plugins logic in a place that isn't in charge of it. You should rather call `LoadAssemblies()` from `PluginsCore` and then evaluate locations of those, it'd also simplify your code.

Reply:
Not sure if using `LoadAssemblies()` again is a good idea, but we can definitely use the following snippet (or one similar to it):

```csharp
PluginsCore.ActivePlugins.Select(static plugin => plugin.GetType().Assembly.Location).ToHashSet()
```

Reply:
> You're duplicating plugins logic in a place that isn't in charge of it. You should rather call `LoadAssemblies()` from `PluginsCore` and then evaluate locations of those, it'd also simplify your code.

@JustArchi @Abrynos That's couse errors on release (I already do it originally but all release check fail due to this [ref [5d7d9ac](https://github.com/JustArchiNET/ArchiSteamFarm/pull/2877/commits/5d7d9ac6ae2cf56ee43395d5835817f81dfd1237)])
Here's error: `'System.Reflection.Assembly.Location' always returns an empty string for assemblies embedded in a single-file app.` [ref [checks](https://github.com/JustArchiNET/ArchiSteamFarm/actions/runs/4664561151/jobs/8256942385)]

Reply:
Then disable that analyzer for this piece of code because we KNOW plugins are not embedded in our single-file app. That's the whole point of being a plugin.

Reply:
For the same reason I said above to use the `!` operator instead of the null coalescing one after calling `Directory.GetParent(string)`. We KNOW plugins are not loaded from a root directory of the filesystem.

Reply:
> Then disable that analyzer for this piece of code because we KNOW plugins are not embedded in our single-file app. That's the whole point of being a plugin.

Ok, how can I suppress this error, I'm not familiar with your github actions.
Also, @JustArchi, is this a good solution?

Reply:
It's good solution, if there is no better way to fetch assembly's location, add IL3000 along other warnings-not-as-errors in Directory.Build.props (it was either there or in ASF csproj, but I'd say there).

Basically I want `ILxxxx` warnings to stay because they can be useful, but we selectively not make them errors when we determine they're not meant to be.